src/wallet/rpcwallet.cpp

```diff
@@ -3792,7 +3792,11 @@ static UniValue getaddressesbylabel(const JSONRPCRequest& request)
     UniValue ret(UniValue::VOBJ);
     for (const std::pair<const CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {
         if (item.second.name == label) {
-            ret.pushKV(EncodeDestination(item.first), AddressBookDataToJSON(item.second, false));
+            UniValue address(UniValue::VOBJ);
+            address.pushKV(EncodeDestination(item.first), AddressBookDataToJSON(item.second, false));
```

Comment:
While `item.first` might be unique, `EncodeDestination()` of it might not be unique. You could work around that by creating a temporary `map<string,type(item.second)>`, fill it and then convert to univalue. It would still be cheaper, since insertion into a map is faster O(log(N)) than insertion into an univalue O(N)

Reply:
Got it, and also would allow to reduce wallet lock scope.

However what happens with duplicate results of `EncodeDestination()` ?

Reply:
in univalue they'd be replaced, in std::map::insert, they'd be skipped

Reply:
Done.

I could reverse the iteration to keep the behaviour, WDYT?